ChopChop/Models/Recipe/Session/SessionRecipe.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+/// Conforms to ActionTimeTracker so that SessionRecipeStep can modify only 1 attribute through the delegate
+class SessionRecipe: ActionTimeTracker {
+    private(set) var timeOfLastAction = Date()
+    private(set) var recipe: Recipe
+    private(set) var sessionSteps: [SessionRecipeStep]! // implicitly unwrapped optional due to need to bind to self
+
+    init(recipe: Recipe) {
+        // A copy of the recipe object is made so that recipe on recipe tab dosen't actually get modified
+        guard let recipeCopy = recipe.copy() as? Recipe else {
+            fatalError("Could not copy recipe for edit")
+        }
+        self.recipe = recipeCopy
+        sessionSteps = recipeCopy.steps.map { SessionRecipeStep(step: $0, actionTimeTracker: self) }
```

Comment:
I think we agreed on this but it looks a little iffy, not sure if there is a better way.

Reply:
Yea I don't really like the implicitly unwrapped optional
I could make actionTimeTracker a class instead, which has no relation to SessionRecipe. so each SessionRecipeStep will have a reference to the same actionTimeTracker and will update the date in the actionTimeTracker [kind of like a shared memory region for the different sessionrecipesteps)

```
class ActionTimeTracker {
    var timeOfLastAction: Date
    func updateTimeOfLastAction(date: Date)
}
```